### PR TITLE
Move Travis secrets from .travis.yml to job config

### PR DESCRIPTION
**Pull request by tgeens**
Builds now runs on travis-ci.com instead of travis-ci.org
Cleaned and updated the `.travis.yml` build file

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
-    before_install:
-    - chmod +x ./gradlew
-    script:
-    - if [[ $TRAVIS_BRANCH = "master" ]] ; then ./gradlew pushDockerImage ; else ./gradlew check --info ; fi
+
+script: ./gradlew buildDockerImage check --info
+
+deploy:
+  provider: script
+  script: ./gradlew buildDockerImage
```
Comment (vierbergenlars) [inline .travis.yml]: You'll want to add `skip_cleanup: true` here, so you can use the gradle buildcache and not rebuild everything.

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-    jdk: oraclejdk8
-    env:
-    - DOCKER_USER=xenitpublisher
-    - secure: "SbStsK3LWwFjDDKhWrbvxAfyk4hSIZctGpl8s0Z3bElSKND6v8bdwNWM27v4ihb5VJeRfslGyzWbP6eUZdbWqTTl/+Pv19oVsRxZQZRmkszpT/xVVAnW6N1+cjypwWWjIt5pX7cQyUfGo4qq2E7iZajApqRZyJLT0luLmHEDiPyhwjlc11xDaplBeBCedoVJ3aX5bgMB1pbXNLmPiAP1TLHEuM0kR1+re7OGqfbskU/eMObA5BV0TH6SburpSoLDNQldXzUbLwwG3ek+LxF4IkfJF05lUdEd/WWNTVHbK5SNh2f3YgptzCUTMr5t06eUxI2JJe3ixGTQZ1ln0eNtUxmm7Z8WUNMjdbMoGB1s58syYtON36a9X8r5WfB2dhA204GnOBwgLvKtLX7TaEgdnSBoBRGqAx7K33N65aHbgc7KgpGhWq0gJJaetEhuKquqyjVL6PZU7m0l/pTwUB5Vnk4BTU9QedvVClee7Jjk+dMV44rmlffQ35TKetvzY4lmflZyo95cdlTBz9mmqyNAdbeg955N0Ll83AdkLNOdvgeHIq7Te2dVuyI95Je1rGTp6+vw/12Vg4e4Esr4uJPJ1FSAFqyIooYqYlOcYuJH4JNuOgnc9mG7V8GaRnRXvO++yB5tTS0XA6HJYH2JIFDMFgvmWXP/JAhfJcEjT4jxy9c="
-    before_install:
-    - chmod +x ./gradlew
-    script:
-    - if [[ $TRAVIS_BRANCH = "master" ]] ; then ./gradlew pushDockerImage ; else ./gradlew check --info ; fi
+
+script: ./gradlew buildDockerImage check --info
```
Comment (vierbergenlars) [inline .travis.yml]: The check task already depends on buildDockerImage.

**Review by vierbergenlars — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (tgeens) [reply .travis.yml]: That was kind of intentional, to "document" what the main thing is we're building here. Maybe not that interesting after all.

File: .travis.yml (hunk 1 above)
Comment (tgeens) [reply .travis.yml]: It should have been `pushDockerImage` too :eyes:

**Review by vierbergenlars — APPROVED** [review]

**Pull request merged**